packages/app/src/runner/studio/AssertionOptions.ce.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <div
+    ref="popper"
+    class="assertion-options"
+  >
+    <div
+      v-for="{ name, value } in options"
+      :key="`${name}${value}`"
+      class="assertion-option"
+      @click.stop="() => onClick(name, value)"
+    >
+      <span
+        v-if="name"
+        class="assertion-option-name"
+      >
+        {{ truncate(name) }}:{{ ' ' }}
+      </span>
+      <span
+        v-else
+        class="assertion-option-value"
+      >
+        {{ truncate(value) }}
+      </span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { createPopper } from '@popperjs/core'
+import { onMounted, ref, nextTick, Ref } from 'vue'
+
+const props = defineProps<{
+  type: string
+  addAssertion: any
+  options: any
```

Comment:
Is there any chance we can go find the correct types for this? If we are making a refactor with the goal of improving code quality, I really feel like we want to be as strict and tight on the type definitions are possible. We know what `options` can be, it should be `have.text.` etc? Those are all defined here https://github.com/cypress-io/cypress/blob/develop/packages/app/src/store/studio-store.ts#L61-L94

Reply:
I re-read the code and defined types.